src/components/templates/PosterPage/usePosterVideo.ts

```diff
@@ -0,0 +1,82 @@
+import { useCallback, useMemo } from "react";
+import { LocalParticipant, RemoteParticipant } from "twilio-video";
+
+import { withId, WithId } from "utils/id";
+
+import { User } from "types/User";
+
+import { useUser } from "hooks/useUser";
+import { useWorldUsersByIdWorkaround } from "hooks/users";
+import { useVideoRoomState } from "hooks/twilio";
+
+export const usePosterVideo = (venueId: string) => {
+  const { userId } = useUser();
+  const { worldUsersById } = useWorldUsersByIdWorkaround();
+
+  const { participants, turnVideoOff, turnVideoOn } = useVideoRoomState({
+    userId,
+    roomName: venueId,
+    showVideoByDefault: false,
+  });
+
+  const getUserById = useCallback(
+    (id: string) => {
+      const user = worldUsersById[id];
+
+      if (!user) return;
+
+      return withId(user, id);
+    },
+    [worldUsersById]
+  );
+
+  const { passiveListeners, activeParticipants } = useMemo(
+    () =>
+      participants.reduce<{
+        passiveListeners: WithId<User>[];
+        activeParticipants: (RemoteParticipant | LocalParticipant)[];
+      }>(
+        (acc, participant) => {
+          // If participant is not broadcasting video, put him into passiveListeners
+          if (participant.videoTracks.size === 0) {
+            const user = getUserById(participant.identity);
+
+            if (!user) return acc;
+
+            return {
+              ...acc,
+              passiveListeners: [...acc.passiveListeners, user],
+            };
+          }
```

Comment:
Should we be choosing between active/passive based on `participant.videoTracks.size`?

How does a user take one of the 'free video slots' to 'join in'? Would it not be better to directly use that same state to decide if they are active/passive?

Reply:
Currently a user has the option to take a seat & temporarily disable the video(if a presenter has some ).

> How does a user take one of the 'free video slots' to 'join in'? Would it not be better to directly use that same state to decide if they are active/passive?

he passes the `hasVideo` to RoomVideoState. When he does that, his camera launches and he passes the videoTrack, which is accessible by all participants

Reply:
So we're not explicitly keeping track of 'seats'/who has sat where/etc?

Reply:
> So we're not explicitly keeping track of 'seats'/who has sat where/etc?

No, the state is kept within twillio itslef

Reply:
_nods_ Fair enough. Not sure that is an ideal implementation/architectural choice here. But if it seems to work, we can deal with that in future.